bottlerocket-settings-models/modeled-types/src/shared.rs

```diff
@@ -1327,3 +1327,76 @@ mod test_kernel_cpu_set_value {
         }
     }
 }
+
+// =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+
+/// SizeString represents a string that contains a valid size specification with optional units.
+/// Valid formats include: "8b", "16kb", "64mb", "1gb", "1.5gb", "100", etc.
+/// The regex pattern is case-insensitive and does not allow any whitespace.
```

Comment:
It's unfortunate that [SOCI uses nonstandard terminology](https://github.com/awslabs/soci-snapshotter/blob/8d8d1200b8eba2daf31a95ce8585a44a553d91cc/config/parallel.go#L29) - what they call "gb" is really "gib", etc.

It'd be worth looking at the `byte_unit` crate to parse the values in a standard way, and then turning the result into an i64 that we would write into the SOCI config (as a string value like it expects).

I would also really like to accept integers in the TOML settings for this, not just strings. The `OciDefaultsResourceLimitV1` struct has a custom deserializer that allows this, and that implementation could be copied and adapted to use `byte_unit` for its string parsing.